tests/meta-mender-ci/recipes-testing/mender-coverage-client/mender-client_%.bbappend

```diff
@@ -9,5 +9,18 @@ do_compile_prepend() {
     test -e ${B}/src/${GO_IMPORT} || ln -s ${S} ${B}/src/${GO_IMPORT}
     cd ${B}/src/${GO_IMPORT}
 
+    # The coverage functionality can only be applied on a clean repo
+    git reset --hard HEAD
```

Comment:
Wow, risky! If you've got externalsrc set, you risk erasing all of your work every time you do a Yocto build!

A better approach would be to copy the directory instead of symlinking it. Then you get a `${B}` folder which is totally independent, and you can do whatever you want there. This also means you don't actually need to call `git reset --hard`, because it will be re-copied every time.

Reply:
haha, yeh, not planning on keeping it like this for production :p 

On the bright side. This worked, and all regular acceptance tests passed: https://northern.tech.gitlab.io/-/Mender/mender-qa/-/jobs/510449468/artifacts/report_accep_qemux86_64_bios_grub.html
except for the versioned clients (which do not have this functionality as of yet).

Which begs the question, should all the released clients have this functionality present also?  Which means the PR to Mender has to be cherry picked.

Or should we change the recipe here to only build on master for now?

Reply:
Hehe, sorry, I think I was a bit overeager, you had not asked for a review yet!

> Which begs the question, should all the released clients have this functionality present also? Which means the PR to Mender has to be cherry picked.
> 
> Or should we change the recipe here to only build on master for now?

No no, master only.

Reply:
> On the bright side. This worked, and all regular acceptance tests passed: 

Nice! :-D